src/utils/attempt.js

```diff
@@ -0,0 +1,11 @@
+// Given a `fallback` and a `preferred` function,
+// return a function that attempts to call the `preferred`.
+// If it fails, or returns undefined, call the fallback and return its value.
+export default function attempt ( fallback, preferred ) {
+	return function ( ...args ) {
```

Comment:
doesn't this mean that later results take priority over earlier ones? Surely we want to use the first successful result? If I do...

```js
one = () => 1;
two = () => 2;
fallback = () => 'fallback';

fn = [ one, two ].reduce( attempt, fallback );
fn().then( result => console.log( result ) );
```

...it logs `2`, when I'd want it to log `1`. What if it was this instead?

```js
function first ( candidates ) {
  return function ( ...args ) {
    return candidates.reduce( ( promise, candidate ) => {
      return promise.then( result => result || Promise.resolve( candidate( ...args ) ) );
    }, Promise.resolve() );
  }
}
```